### PR TITLE
allow seperate learning rate "muon_lr" and "adam_lr" for muon optimizer

### PR DESCRIPTION
**Pull request by delock**
This PR allows seperate learning rate for muon and adam part of the Muon optimizer.  Following up https://github.com/deepspeedai/DeepSpeed/issues/7657

### CODE REVIEW

File: deepspeed/runtime/engine.py [hunk 1]
```diff
@@ -1599,15 +1599,21 @@ def _configure_basic_optimizer(self, model_parameters):
             param_groups = []
             if muon_params:
                 accepted_parameters = dict()
-                for key in ["lr", "momentum", "weight_decay"]:
+                for key in ["lr", "momentum", "weight_decay", "muon_lr"]:
```
Comment (PKUWZP) [inline deepspeed/runtime/engine.py:1602]: Does the user need to specific "muon_lr" for their training jobs?

**Review by PKUWZP — APPROVED** [review]

File: deepspeed/runtime/engine.py (hunk 1 above)
Comment (delock) [reply deepspeed/runtime/engine.py:1602]: Not mandatory, 'lr' will apply to both muon and adam.  However, 'muon_lr' will be used if present.  Same for 'adam_lr'.  Thinking them as advanced user setting.

**Comment by delock** [comment]
@PKUWZP @tjruwase I'll put this PR into auto-merge if everything looks fine.
